compiler/src/iree/compiler/Codegen/Common/GPU/GPUNestedLayoutDistributionPatterns.cpp

```diff
@@ -123,36 +122,18 @@ getElementVectorTileShape(NestedLayoutAttr vectorLayout) {
 /// Computes the warp and thread indices for the given vector layout from a
 /// single linearized thread ID.
 static void populateWarpAndThreadIndices(RewriterBase &rewriter, Value threadId,
+                                         int64_t subgroupSize,
                                          NestedLayoutAttr vectorLayout,
                                          SmallVector<Value> &warpIndices,
                                          SmallVector<Value> &threadIndices) {
-  int64_t subgroupRank = vectorLayout.getSubgroupBasis().size();
   // The delinearized thread IDs are returned from outer most to inner most,
   // i.e. before applying the layout described dimensions ordering.
+  int64_t rank = vectorLayout.getRank();
   SmallVector<Value> threadIds =
-      vectorLayout.computeThreadIds(threadId, rewriter);
-
-  SmallVector<Value> filteredSubgroupIds;
-  for (auto [id, active] :
-       llvm::zip(threadIds, vectorLayout.getSubgroupActiveIds())) {
-    if (active)
-      filteredSubgroupIds.push_back(id);
-  }
-  SmallVector<Value> filteredThreadIds;
-  for (auto [id, active] : llvm::zip(llvm::drop_begin(threadIds, subgroupRank),
-                                     vectorLayout.getThreadActiveIds())) {
-    if (active)
-      filteredThreadIds.push_back(id);
-  }
-
-  // Subgroup and thread (lane) indices normalized to the order in which
-  // they are used by each dimension.
-  warpIndices = llvm::to_vector(
-      llvm::map_range(invertPermutationVector(vectorLayout.getSubgroupOrder()),
-                      [&](int64_t i) { return filteredSubgroupIds[i]; }));
-  threadIndices = llvm::to_vector(
-      llvm::map_range(invertPermutationVector(vectorLayout.getThreadOrder()),
-                      [&](int64_t i) { return filteredThreadIds[i]; }));
+      vectorLayout.computeThreadIds(threadId, subgroupSize, rewriter);
+  warpIndices = SmallVector<Value>(threadIds.begin(), threadIds.begin() + rank);
```

Comment:
I know it's slightly less efficient, but any thoughts on using std::pair<SmallVector<Value>> or a struct, for a cleaner interface?

Reply:
yeah... maybe in some other patch, this one is already heavy